[BUG] test_array_element_at_zero_index_fail test failures in Spark 3.4

**Describe the bug**
This is related to #8117. Since the exception actually was changed during the release candidate cycle for Spark 3.4, there are a few integration test failures that need to be resolved in addition to compile error that was fixed:

```
FAILED ../../src/main/python/array_test.py::test_array_element_at_zero_index_fail[True-0] - AssertionError: Expected error 'org.apache.spark.SparkRuntimeException: [E...
FAILED ../../src/main/python/array_test.py::test_array_element_at_zero_index_fail[True-Integer] - AssertionError: Expected error 'org.apache.spark.SparkRuntimeExcepti...
FAILED ../../src/main/python/array_test.py::test_array_element_at_zero_index_fail[False-Integer][INJECT_OOM] - AssertionError: Expected error 'org.apache.spark.SparkR...
FAILED ../../src/main/python/array_test.py::test_array_element_at_zero_index_fail[False-0][INJECT_OOM] - AssertionError: Expected error 'org.apache.spark.SparkRuntime...
```

```
E       AssertionError: Expected error 'org.apache.spark.SparkRuntimeException: [ELEMENT_AT_BY_INDEX_ZERO] The index 0 is invalid' did not appear in 'org.apache.spark.SparkException: Job aborted due to stage failure: Task 2 in stage 26.0 failed 1 times, most recent failure: Lost task 2.0 in stage 26.0 (TID 106) (192.168.4.31 executor driver): org.apache.spark.SparkRuntimeException: [INVALID_INDEX_OF_ZERO] The index 0 is invalid. An index shall be either < 0 or > 0 (the first element has index 1).
E       	at org.apache.spark.sql.errors.QueryExecutionErrors$.invalidIndexOfZeroError(QueryExecutionErrors.scala:1614)
E       	at org.apache.spark.sql.errors.QueryExecutionErrors.invalidIndexOfZeroError(QueryExecutionErrors.scala)
E       	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
E       	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
E       	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
E       	at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
E       	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:888)
E       	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:888)
E       	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
E       	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:364)
E       	at org.apache.spark.rdd.RDD.iterator(RDD.scala:328)
E       	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:92)
E       	at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:161)
E       	at org.apache.spark.scheduler.Task.run(Task.scala:139)
E       	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:554)
E       	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1529)
E       	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:557)
E       	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
E       	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
E       	at java.lang.Thread.run(Thread.java:750)
```

**Steps/Code to reproduce bug**

Run the specified integration tests:
```
cd spark-rapids/integration_tests/
./run_pyspark_from_build.sh -k test_array_element_at_zero_index_fail
```

--- a/integration_tests/src/main/python/array_test.py
+++ b/integration_tests/src/main/python/array_test.py
@@ -262,9 +262,12 @@ def test_array_element_at_ansi_not_fail_all_null_data():
 @pytest.mark.parametrize('index', [0, array_zero_index_gen], ids=idfn)
 @pytest.mark.parametrize('ansi_enabled', [False, True], ids=idfn)
 def test_array_element_at_zero_index_fail(index, ansi_enabled):
-    message = "SQL array indices start at 1" \
-        if is_before_spark_340() and not is_databricks113_or_later() else \
-        "org.apache.spark.SparkRuntimeException: [ELEMENT_AT_BY_INDEX_ZERO] The index 0 is invalid"
+    if is_spark_340_or_later():
+        message = "org.apache.spark.SparkRuntimeException: [INVALID_INDEX_OF_ZERO] The index 0 is invalid"
+    elif is_databricks113_or_later():
+        message = "org.apache.spark.SparkRuntimeException: [ELEMENT_AT_BY_INDEX_ZERO] The index 0 is invalid"
+    else:
+        message = "SQL array indices start at 1"
 
     if isinstance(index, int):
         test_func = lambda spark: unary_op_df(spark, ArrayGen(int_gen)).select(

